### ISSUE TITLE
The performance gap between the val set and test set

### ISSUE THREAD

**Issue by PkuRainBow** [mapillary/inplace_abn#25]
Hi, I have trained some models and achieved new state-of-the-art performance on the val set of the cityscapes.( the modified ResNet)

But when I submit the results on the test set. I find the performance gap is very large! 

With single crop, the performance on the val set will drop 2 points on the test set. I guess maybe it is related to the DSN structure or related to the learning rate  and so on.

Have you ever meet such problem, we can achieve 80.3 on the val set with single crop. But we can only achieve 78.0 on the test set with single crop......

Which is very frustrating........ Our previous methods could achieve 78.7 on val set and 77.7 on test set, where the performance gap is 1 point.

However, my previous methods without DSN only suffers 1 point drop between the val set and test set.

Could you help me on how to decrease this gap?

Difference with my previous method:

```
backbone: resnet -> modified resnet (replace the 7x7 conv with three 3x3 convs)
learning rate: 7e-3 -> 2e-2
pretrain: COCO pretrain -> ImageNet pretrain.
loss: no extra loss ->deeply supervisde loss

```
There exist some differences, I just want to ask advice how to avoid this problem.

**Comment by PkuRainBow** [comment]
@dongzhuoyao Are you using this baseline? How about your learning rate, weight decay and backbone?

**Comment by PkuRainBow** [comment]
Maybe this gap is related to the COCO pretraining? Or related to the modified structure of the resnet?

**Comment by dongzhuoyao** [comment]
I think this is not related to learning rate, weight decay and backbone. Maybe there is a problem in your local mIoU evaluation code.

**Comment by PkuRainBow** [comment]
@dongzhuoyao So do you mean that you have find some bugs in your local evaluation code?

**Comment by PkuRainBow** [comment]
@dongzhuoyao I find that my performance drop is closely related to the class truck.
The mIOU for truck is 81.6 on val set and drops to 62 on the test set. It is very strange.

**Comment by ducksoup** [comment]
I'll close this issue since it doesn't appear to be related with the InPlace-ABN code.
